[caf] update to <1.0.1>

### Library name

caf

### New version number

1.0.1

### Other information that may be useful (release notes, etc...)

current version 0.19.6 had something wrong
environment:
windows 10
visual studio 2022
c++ language standard: c++20

i have installed caf with vcpkg and its integrated with visual studio

vcpkg install caf

getting several errors, given part of the error here:
1>D:\vcpkg\installed\x64-windows\include\caf\message.hpp(195,12): warning C4251: 'caf::message::data_': class 'caf::intrusive_cow_ptr<caf::detail::message_data>' needs to have dll-interface to be used by clients of class 'caf::message'
1>(compiling source file 'hello.cpp')
1>D:\vcpkg\installed\x64-windows\include\caf\intrusive_cow_ptr.hpp(37,7):
1>see declaration of 'caf::intrusive_cow_ptr<caf::detail::message_data>'
1>D:\vcpkg\installed\x64-windows\include\caf\error.hpp(211,25): warning C4251: 'caf::error::data_': class 'std::unique_ptr<caf::error::data,std::default_delete<caf::error::data>>' needs to have dll-interface to be used by clients of class 'caf::error'
1>(compiling source file 'hello.cpp')
1>C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33519\include\memory(3172,7):
1>see declaration of 'std::unique_ptr<caf::error::data,std::default_delete<caf::error::data>>'
1>D:\vcpkg\installed\x64-windows\include\caf\chrono.hpp(103,22): warning C4251: 'caf::chrono::datetime::utc_offset': class 'std::optional<int>' needs to have dll-interface to be used by clients of class 'caf::chrono::datetime'
1>(compiling source file 'hello.cpp')
1>C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33519\include\optional(213,7):
1>see declaration of 'std::optional<int>'
1>D:\vcpkg\installed\x64-windows\include\caf\optional.hpp(272,22): error C4996: 'caf::optional<T>': use std::optional instead
1>(compiling source file 'hello.cpp')
1>D:\vcpkg\installed\x64-windows\include\caf\optional.hpp(22,7):
1>see declaration of 'caf::optional'
1>D:\vcpkg\installed\x64-windows\include\caf\optional.hpp(280,22): error C4996: 'caf::optional<T>': use std::optional instead

--- a/ports/caf/fix_cxx17.patch
+++ b/ports/caf/fix_cxx17.patch
@@ -1,40 +1,20 @@
 diff --git a/CMakeLists.txt b/CMakeLists.txt
-index f088f6a..a8bcf80 100644
+index 0622702..f44e0f5 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -134,7 +134,7 @@ endif()
- 
- if(NOT DEFINED CAF_USE_STD_FORMAT)
-   set(CAF_USE_STD_FORMAT OFF CACHE BOOL "Enable std::format support" FORCE)
--  if(NOT CMAKE_CROSSCOMPILING)
-+  if(0)
-     set(snippet "#include <format>
-                  #include <iostream>
-                  int main() { std::cout << std::format(\"{}\", \"ok\"); }")
-@@ -180,8 +180,6 @@ endif()
+@@ -158,7 +158,6 @@ install(TARGETS caf_internal EXPORT CAFTargets)
  
  # -- create the libcaf_test target ahead of time for caf_core ------------------
  
 -add_library(libcaf_test)
--
+ 
  # -- add uninstall target if it does not exist yet -----------------------------
  
- if(NOT TARGET uninstall)
-@@ -326,8 +324,6 @@ function(caf_add_component name)
-     list(APPEND targets ${tst_bin_target})
-     add_executable(${tst_bin_target}
-                    ${CAF_ADD_COMPONENT_LEGACY_TEST_SOURCES})
--    target_link_libraries(${tst_bin_target} PRIVATE libcaf_test
--                          ${CAF_ADD_COMPONENT_DEPENDENCIES} ${lib_target})
-     target_include_directories(${tst_bin_target} PRIVATE
-                                "${CMAKE_CURRENT_SOURCE_DIR}/tests/legacy")
-     if(CAF_ADD_COMPONENT_LEGACY_TEST_SUITES)
-@@ -383,8 +379,6 @@ endfunction()
+@@ -351,7 +350,6 @@ endfunction()
  
  add_subdirectory(libcaf_core)
  
 -add_subdirectory(libcaf_test)
--
+ 
  if(CAF_ENABLE_NET_MODULE)
    add_subdirectory(libcaf_net)
- endif()

--- a/ports/caf/portfile.cmake
+++ b/ports/caf/portfile.cmake
@@ -2,8 +2,8 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO actor-framework/actor-framework
     REF "${VERSION}"
-    SHA512 8a7aacbd9bf18318d9ca1f5fb30c101220c1eef2c4bfe82c53760024022473109038872c0deb5a60d732a91da8d863c556a27018e6b667bfcfbf536df3cebcaf
-    HEAD_REF master
+    SHA512 0849a0b17a2c011142dd56ff88b01d764dc57b40de5d8467bcf1aa2ddd7540415228dc05f8652b9534e18fb8c2e28465bf881fe1e1bf5c7f9919cfb3edd573d3
+    HEAD_REF main
     PATCHES
         fix_dependency.patch
         fix_cxx17.patch
@@ -12,7 +12,6 @@ vcpkg_from_github(
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
-        -DCMAKE_DISABLE_FIND_PACKAGE_Doxygen=ON
         -DCAF_ENABLE_CURL_EXAMPLES=OFF
         -DCAF_ENABLE_PROTOBUF_EXAMPLES=OFF
         -DCAF_ENABLE_QT6_EXAMPLES=OFF
@@ -20,10 +19,8 @@ vcpkg_cmake_configure(
         -DCAF_ENABLE_ACTOR_PROFILER=OFF
         -DCAF_ENABLE_EXAMPLES=OFF
         -DCAF_ENABLE_TESTING=OFF
-        -DCAF_ENABLE_TOOLS=OFF
         -DCAF_ENABLE_IO_MODULE=ON
         -DCAF_ENABLE_EXCEPTIONS=ON
-        -DCAF_ENABLE_UTILITY_TARGETS=OFF
 )
 
 vcpkg_cmake_install()
@@ -33,5 +30,6 @@ vcpkg_cmake_config_fixup(PACKAGE_NAME CAF CONFIG_PATH lib/cmake/CAF)
 vcpkg_copy_pdbs()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include" "${CURRENT_PACKAGES_DIR}/debug/share")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/include/caf/internal")
 
 vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")

--- a/ports/caf/vcpkg.json
+++ b/ports/caf/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "caf",
-  "version": "0.19.6",
+  "version": "1.0.1",
   "description": "an open source implementation of the actor model for C++ featuring lightweight & fast actor implementations, pattern matching for messages, network transparent messaging, and more.",
   "homepage": "https://github.com/actor-framework/actor-framework",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -1469,7 +1469,7 @@
       "port-version": 0
     },
     "caf": {
-      "baseline": "0.19.6",
+      "baseline": "1.0.1",
       "port-version": 0
     },
     "caffe2": {

--- a/versions/c-/caf.json
+++ b/versions/c-/caf.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "2cb6b24903b98d8335f18f4964ce6823550b836b",
+      "version": "1.0.1",
+      "port-version": 0
+    },
+    {
       "git-tree": "e71a51cfec683161a30d2dcca04fe5fc7f063ea7",
       "version": "0.19.6",
       "port-version": 0

